### ISSUE TITLE
has_input annotation extension not showing in table view

### ISSUE THREAD

**Issue by vanaukenk** [geneontology/noctua-form#127]
@tmushayahama @suzialeksander 

In this model:  
http://noctua.geneontology.org/editor/graph/gomodel:5fce9b7300001890?model_id=gomodel:5fce9b7300001890

![image](https://user-images.githubusercontent.com/1730534/104734583-78a26780-570e-11eb-8ea0-37f027ce2f82.png)

there is a 'has input' annotation extension on the first BP for YPQ2 that is not showing in the table view

![image](https://user-images.githubusercontent.com/1730534/104734550-6cb6a580-570e-11eb-9636-74f6a0efada5.png)

and the option to add it, or any extension to that first process, is not available:

![image](https://user-images.githubusercontent.com/1730534/104734500-5dcff300-570e-11eb-9b55-7a7291834792.png)

**Comment by ukemi** [comment]
I wonder if this is the same issue as me not being able to see the errors in ART.

**Comment by suzialeksander** [comment]
Just for clarification, I was only able to add the AE as pictured in GE, the option to add the AE (where I wanted it) never appeared in Form.

**Comment by vanaukenk** [comment]
@suzialeksander 
The option to add a 'has input' is available in the BP only version of the form, but it doesn't appear to be an option in the table view, post-entry right?

**Comment by suzialeksander** [comment]
OK, I was just able to reproduce.  

* start with BP Annotation Form
* add a specific process, like "basic amino acid transmembrane import into vacuole".  
* add a general process, like "cellular response to nitrogen starvation" 
* The option to add a "has input" to the specific process has now disappeared, but I can add the input (L-arginine) to the general cellular response BP I added second.


It looks like if I add the specific process "basic amino acid transmembrane import into vacuole" I can immediately add my desired input, but as soon as I've added a second BP I'm locked out of extensions to the first BP in Form.  In theory this can be worked around by having the complete annotation planned out before starting, but this isn't curator-friendly as in this case the "has input" was a bit of an afterthought during the actual curation of this paper.

**Comment by suzialeksander** [comment]
but yes, the "(...) -> Add" menu option is blank in the table view also (but the table view is all jumpy and just uncooperative in general for me)

**Comment by tmushayahama** [comment]
Thanks @suzialeksander, lemme double check on the bp only form, it has been a while. @vanaukenk  I think on bpOnly form "has input" relation on BP, Noctua Form was not aware of the it. It was only on MF. So after causal edge, it stops reading edges after BP. will fix it soon.

As for the process of adding the extension and getting locked out, its actually a bug, I never thought of. Thanks for reporting. I will take a look, but I think once we fix "has input/output" on BP, things will flow. 

@ukemi I was about to get back to you on the ART shex errors issues

**Comment by tmushayahama** [comment]
> I wonder if this is the same issue as me not being able to see the errors in ART.

and yes thanks, I think that is mostly it. That the individuals and relations on graph are unknown to Noctua Form yet and it was trying to understand the shex violation from reasoner. I will create another ticket on this one and continue from there.

**Comment by vanaukenk** [comment]
@suzialeksander 
we tested this on noctua-dev on the 2023-09-14 workbenches call and it looks to be working fine now.
Please also test if you'd like and let us know if you're happy with it.

**Comment by vanaukenk** [comment]
@suzialeksander 
Let us know if you've had a chance to at least check the 'has input' display on the table on noctua-dev.
It seems to be working fine for me, so we will try to move this fix into production later this week.
